Sources/Ticketmaster/TicketmasterManager.swift

```diff
@@ -55,7 +57,89 @@ class TicketmasterManager {
                 member.value = nil
             }
         }
+        
+        // Begin observing for TM PSDK's events.
+        TicketmasterManager.tmEvents.keys.forEach { notificationName in
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(self.receiveTicketmasterNotification),
+                name: NSNotification.Name(rawValue: notificationName),
+                object: nil
+            )
+        }
     }
+    
+    @objc
+    func receiveTicketmasterNotification(_ notification: Foundation.Notification) {
+        guard let roverScreenName = TicketmasterManager.tmEvents[notification.name.rawValue] else {
+            os_log("TicketmasterManager received an unexpected NSNotification, ignoring.", log: .general, type: .error)
+            return
+        }
+        
+        // the same fields are common amongst all the events we monitor for.
+        let attributes: Attributes = [:]
+        
+        let eventAttributes: Attributes = [:]
+        let venueAttributes: Attributes = [:]
+        let artistAttributes: Attributes = [:]
+        
+        if let eventID = notification.userInfo?["event_id"] {
+            eventAttributes["id"] = eventID
+        }
+        
+        if let eventName = notification.userInfo?["event_name"] {
+            eventAttributes["name"] = eventName
+        }
+
+        if let eventDate = notification.userInfo?["event_date"] as? Date {
+            // In order to be (somewhat) consistent with the Android version of the Presence SDK (which yields a pre-rendered date string in a non-standard format), render it into the following format:
+            // Mon, Apr 13, 7 PM
+            let formatter = DateFormatter()
+            formatter.locale = Locale(identifier: "en_US_POSIX")
+            formatter.setLocalizedDateFormatFromTemplate("E, MMM d, h:mm a")
```

Comment:
I guess this will occasionally result in some differences due to the `:mm` part?

Reply:
that's right, although the android sdk does indeed include those minutes if they are present (ie., not 0).

Reply:
clarified the comment a bit.